oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreMBeanImpl.java

```diff
@@ -264,4 +268,28 @@ public String cleanIndividualCache(String name) {
                 return "ERROR: Invalid cache name received.";
         }
     }
+
+    @Override
+    public String createCheckpoint(String revision, long lifetime, boolean force) {
+        Revision rev = Revision.fromString(revision);
+        long oldestTimestamp = nodeStore.getClock().getTime() - revisionGCMaxAgeMillis;
+        Revision oldestCheckpoint = nodeStore.getCheckpoints().getOldestRevisionToKeep();
+        if (oldestCheckpoint != null) {
+            oldestTimestamp = Math.min(oldestTimestamp, oldestCheckpoint.getTimestamp());
+        }
+        if (force || oldestTimestamp < rev.getTimestamp()) {
+            Map<String, String> info = new HashMap<>();
+            info.put(COMPOSITE_INFO + "created", Long.toString(rev.getTimestamp()));
+            info.put(COMPOSITE_INFO + "expires", Long.toString(Utils.sum(rev.getTimestamp() + lifetime)));
+
+            String cp = nodeStore.getCheckpoints().create(lifetime, info, rev).toString();
+            log.info("Created checkpoint [{}] with lifetime {} for Revision {}", cp, lifetime, revision);
+            return cp;
```

Comment:
I was thinking about returning here the string inside the log.info, instead of the "cp" variable that just returns the revision string (which is the first argument you send to this method, so not really useful).
But that would be less machine readable, and I'm not sure if that would be needed.

Reply:
Something like:
return String.format("Created checkpoint [{}] with lifetime {} for Revision {}", cp, lifetime, revision);

But changing this would mean at least 2 tests would need to be adjusted:
createCheckpointForceWithRevisionTooOld()
createCheckpointBeforeExistingCheckpoint()

Reply:
Thanks for the suggestion. I merged your PR.